### PR TITLE
Homework №5. User stuff.

### PR DESCRIPTION
**Pull request by kkulahin**
### _Homework №5_

- Describe base class Entity that can store name​

- Implement child classes Stuff, Box, User. Box can store some stuff[]. Every box belongs to user​

- Add example of usage of these classes, e.g., create some instances, display data related to them​

- Push code to repo “user stuff”

image

### CODE REVIEW

File: hw_5/script.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  constructor(name) {
+    super(name);
+  }
+
+  toString() {
+    return `{ User name: ${this.name} }`;
+  }
+}
+
+// let kanstantsin = new User("Kanstantsin");
```
Comment (MishkanJigan) [inline hw_5/script.js]: Use single сomment for description and a multi-line сomment for the code we can use, so it is clearer.

```suggestion
// Example of usage

/*
let kanstantsin = new User("Kanstantsin");

let kanstantinsFirstBox = new Box("box1", kanstantsin);

let kanstantinsSecondBox = new Box("box2", kanstantsin, [new Stuff("pen")]);

console.log(kanstantinsSecondBox);

kanstantinsSecondBox.addStuff(new Stuff("credit card"));

console.log(kanstantinsSecondBox + "");

try {
  kanstantinsSecondBox.addStuff("phone");
} catch (e) {
  console.error(e.message);
}

try {
  kanstantinsFirstBox.owner = new User("Alisa");
  console.log(kanstantinsFirstBox.owner);
  kanstantinsFirstBox.owner = "Peter";
} catch (e) {
  console.error(e.message);
}

console.log(kanstantinsFirstBox + "");

console.log(new Entity("entity").toString());
*/
```

File: hw_5/script.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+//
+// let kanstantinsFirstBox = new Box("box1", kanstantsin);
```
Comment (MishkanJigan) [inline hw_5/script.js]: can we use here `const `insted `let`?

File: hw_5/script.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+//
+// let kanstantinsSecondBox = new Box("box2", kanstantsin, [new Stuff("pen")]);
+//
+// console.log(kanstantinsSecondBox);
+//
+// kanstantinsSecondBox.addStuff(new Stuff("credit card"));
+//
+// console.log(kanstantinsSecondBox + "");
```
Comment (MishkanJigan) [inline hw_5/script.js]: Why you use `+""` in this kind of console.log?

Comment (kkulahin) [reply hw_5/script.js]: > Why you use `+""` in this kind of console.log?

To initiate a call `toString()` method from the `Box` class.

https://github.com/BlackTruth/javascript/blob/45921bc3f500ef5300cc1b99c0452a14d4a9b84a/hw_5/script.js#L39-L41

File: hw_5/script.js (hunk 1 above)
Comment (kkulahin) [reply hw_5/script.js]: https://github.com/BlackTruth/javascript/blob/3e02ee78527ee2ff262c41e206de768fde8c4212/hw_5/script.js#L62-L76
Fixed.

File: hw_5/script.js (hunk 2 above)
Comment (kkulahin) [reply hw_5/script.js]: https://github.com/BlackTruth/javascript/blob/3e02ee78527ee2ff262c41e206de768fde8c4212/hw_5/script.js#L63-L67
I don't pay attention to it as always) Yes, we can, I fixed it.

**Pull request merged**